src/fastapi_cloud_cli/commands/deploy.py

```diff
@@ -350,43 +339,50 @@ def _wait_for_deployment(
     with toolkit.progress(
         next(messages), inline_logs=True, lines_to_show=20
     ) as progress:
-        with handle_http_errors(progress=progress):
-            for line in _stream_build_logs(deployment.id):
-                time_elapsed = time.monotonic() - started_at
+        with APIClient() as client:
```

Comment:
nit: we could combine the `with` statements to reduce the nesting by one layer

```suggestion
with toolkit.progress(
    next(messages), inline_logs=True, lines_to_show=20
) as progress, APIClient() as client:
```